src/main/java/frc/robot/subsystems/swerve/module/Modules.java

```diff
@@ -0,0 +1,148 @@
+package frc.robot.subsystems.swerve.module;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.Arrays;
+
+public class Modules {
+
+	private final Module[] modules;
+	private final String logPath;
+
+	public Modules(String logPath, Module... modules) {
+		this.logPath = logPath + ModuleConstants.MODULES_LOG_PATH_ADDITION;
+		this.modules = modules;
+	}
+
+	public Module getModule(ModuleUtils.ModulePosition modulePosition) {
+		return modules[modulePosition.getIndex()];
+	}
+
+
+	public void updateInputs() {
+		for (Module currentModule : modules) {
+			currentModule.updateInputs();
+		}
+		Logger.recordOutput(logPath + "CurrentStates", getCurrentStates());
+		Logger.recordOutput(logPath + "TargetStates", getTargetStates());
+	}
+
+
+	public void resetModulesAngleByEncoder() {
+		for (Module module : modules) {
+			module.resetSteerByEncoder();
+		}
+	}
+
+	public void setBrake(boolean brake) {
+		for (Module currentModule : modules) {
+			currentModule.setBrake(brake);
+		}
+	}
+
+
+	public void pointWheels(Rotation2d targetSteerPosition, boolean optimize) {
+		for (Module module : modules) {
+			module.pointSteer(targetSteerPosition, optimize);
+		}
+	}
+
+	public void pointWheelsInCircle() {
+		Arrays.stream(modules).forEach(Module::pointInCircle);
+	}
+
+	public void pointWheelsInX() {
+		Arrays.stream(modules).forEach(Module::pointToCenter);
+	}
+
+
+	public void stop() {
+		for (Module currentModule : modules) {
+			currentModule.stop();
+		}
+	}
+
+	public void setSteersVoltage(double voltage) {
```

Comment:
just to make sure - the only time I can think this func is useful is when doing sysid, other then that it has no use, right?

Reply:
drive is useful in open loop drive, but steer is for sysid